librocketchat.c

```diff
@@ -3636,6 +3802,27 @@ rc_add_account_options(GList *account_options)
 	
 	option = purple_account_option_string_new(N_("Server Path"), "server_path", "");
 	account_options = g_list_append(account_options, option);
+
+#define ADD_VALUE(list, desc, v) { \
+	PurpleKeyValuePair *kvp = g_new0(PurpleKeyValuePair, 1); \
+	kvp->key = g_strdup((desc)); \
+	kvp->value = g_strdup((v)); \
+	list = g_list_prepend(list, kvp); \
+}
+
+
+	ADD_VALUE(encryption_values, _("No encryption"), "no_tls");
+	ADD_VALUE(encryption_values, _("Require encryption"), "tls");
+
+#undef ADD_VALUE
+
+	option = purple_account_option_list_new(_("Connection security"),
+                                            "connection_security", encryption_values);
+	account_options = g_list_append(account_options, option);
+
+	option = purple_account_option_string_new(N_("Proxy"), "proxy", "");

```

Comment:
There's proxy options available in the libpurple api that can be used instead from the proxy tab using the `PurpleProxyInfo` stuff in `purple_proxy_get_setup()` https://docs.imfreedom.org/pidgin2/proxy_8h.html#a6995e6f63718dcd8644b9d4ef471392a  if that makes things easier?

Reply:
The thing is that isn't really a "proxy", it's more like using another host instead of the real host.
It works by using either stunnel or socat as an ssl proxy - both work fine from my tests.

Reply:
Oh right!  This is so you can add in the client cert?  Just putting 2 and 2 together now.  With a label like "proxy" I wouldn't want anyone to get tripped up with the other Proxy tab. 

Naming stuff is hard.

What about "intermediate server"?

Reply:
> What about "intermediate server"?

Isn't that also a proxy? I think proxy is accurate.